pkg/controller/testutil/test_utils.go

```diff
@@ -417,6 +419,19 @@ func (f *FakeRecorder) AnnotatedEventf(obj runtime.Object, annotations map[strin
 	f.Eventf(obj, eventtype, reason, messageFmt, args...)
 }
 
+func (f *FakeRecorder) AssertEventReasons(events []v1.Event) bool {
+	f.Lock()
+	defer f.Unlock()
+	equalEventReason := func(a *v1.Event, b v1.Event) bool {
+		if a.Reason != b.Reason {
+			klog.FromContext(context.TODO()).Error(nil, fmt.Sprintf("AssertEventReasons: expected %#v, actually %#v", a.Reason, b.Reason))
```

Comment:
I don't think this is a good place to log errors. It should be done in the test case using assert/require in my opinion. BTW, if lengths of compared slices are different the error message will not be shown.

Reply:
sg, moved logging into the test case, which means error is shown for slices of different lengths. thanks!